### PR TITLE
Adding nexus in ref

### PR DESCRIPTION
**Pull request by RubelMozumder**
Implement Nexus file generation for XRD measurements.

PR #116  handled in #118 and #150.

### CODE REVIEW

File: src/nomad_measurements/xrd/nx.py [hunk 1]
```diff
@@ -37,3 +47,86 @@
...
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_current': 'archive.data.xrd_settings.source.xray_tube_current',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_voltage': 'archive.data.xrd_settings.source.xray_tube_voltage',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one': 'archive.data.xrd_settings.source.kalpha_one',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two': 'archive.data.xrd_settings.source.kalpha_two',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone': 'archive.data.xrd_settings.source.ratio_kalphatwo_kalphaone',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta': 'archive.data.xrd_settings.source.kbeta',
+}
+
+
+def walk_through_object(parent_obj, attr_chain):
```
Comment (ka-sarthak) [inline src/nomad_measurements/xrd/nx.py:69]: There is some common code in this function and `HDF5Handler._set_hdf5_reference`. Maybe we can refactor this later

File: src/nomad_measurements/xrd/nx.py [hunk 2]
```diff
@@ -37,3 +47,86 @@
...
+            else:
+                child_obj = getattr(parent_obj, child_nm, None)
+            if child_obj is None:
+                return None
+            parent_obj = child_obj
+
+        return child_obj
+
+
+def populate_nx_dataset_and_attribute(
```
Comment (ka-sarthak) [inline src/nomad_measurements/xrd/nx.py:102]: This can be made part of the HDF5Handler, no?

File: src/nomad_measurements/xrd/schema.py [hunk 3]
```diff
@@ -1137,6 +1139,11 @@ class ELNXRayDiffraction(XRayDiffraction, EntryData):
     diffraction_method_name.m_annotations['eln'] = ELNAnnotation(
         component=ELNComponentEnum.EnumEditQuantity,
     )
+    nexus_results = Quantity(
+        type=ArchiveSection,
+        description='Reference to the NeXus entry.',
+        a_eln = ELNAnnotation(component='ReferenceEditQuantity')
```
Comment (ka-sarthak) [inline src/nomad_measurements/xrd/schema.py:1145]: Consider using `ELNComponentEnum.ReferenceEditQuantity`

Comment (RubelMozumder) [reply src/nomad_measurements/xrd/schema.py]: Done!

File: src/nomad_measurements/xrd/nx.py (hunk 2 above)
Comment (RubelMozumder) [reply src/nomad_measurements/xrd/nx.py]: Done!

**Review by ka-sarthak — APPROVED** [review]
Looks great!

**Pull request merged**